*** Copy Images and Rename ***
==============================

In [1]:
# Copy Images and Rename
import os 
from shutil import copy2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderPercobaan = r"Percobaan 2 - Atap Kamar (Vertikal)\\"
# "Percobaan 2 - Atap Kamar (Vertikal)\Final\\""
# "Percobaan 1 - Depan Kamar (Horizontal)\\"

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("Folder Percobaan: ", folderPercobaan)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# Function to rename multiple files 
def main(): 
    a,b,c,d = 0,0,0,0

    #srcPath = folderPercobaan
    srcPath = folderPercobaan + r"Final\\"                  # "Percobaan 2 - Atap Kamar (Vertikal)\Final\\""

    #dstPath = folderPercobaan + r"_Result_\\"
    dstPath = folderPercobaan + r"Final\\_Result_\\"

    dstPath2 = r"_Result Calibration_\Test " + str(numberTest) + "\\1_Original-Images\\" + folderPercobaan
      
    for filename in os.listdir(srcPath): 
        # Copy Original Images
        oriSrc = srcPath + filename
        if filename[:10] == Rotate +"Camera_A":
            newFilename = Rotate + "A_"
            a += 1
            newDst = dstPath + newFilename + str(a) + ".jpg"
            newDst2 = dstPath2 + newFilename + str(a) + ".jpg"
        elif filename[:10] == Rotate + "Camera_B":
            newFilename = Rotate + "B_"
            b += 1
            newDst = dstPath + newFilename + str(b) + ".jpg"
            newDst2 = dstPath2 + newFilename + str(b) + ".jpg"
        elif filename[:10] == Rotate + "Camera_C":
            newFilename = Rotate + "C_"
            c += 1
            newDst = dstPath + newFilename + str(c) + ".jpg"
            newDst2 = dstPath2 + newFilename + str(c) + ".jpg"
        elif filename[:10] == Rotate + "Camera_D":
            newFilename = Rotate + "D_"
            d += 1
            newDst = dstPath + newFilename + str(d) + ".jpg"
            newDst2 = dstPath2 + newFilename + str(d) + ".jpg"

        # Copy Original Images
        print("oriSrc: ", oriSrc)
        print("newDst: ", newDst)
        print("newDst2: ", newDst2)
        print()
        copy2(oriSrc, newDst)
        copy2(oriSrc, newDst2)
          
        # rename() function will 
        # rename all the files 
        #os.rename(src, dst) 
  
# Driver Code 
if __name__ == '__main__': 
      
    # Calling main() function 
    main()

Number Test:  1
Rotate True
Folder Percobaan:  Percobaan 2 - Atap Kamar (Vertikal)\\
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

oriSrc:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\R.Camera_A_ 05-Jul-2021_16_13_23.jpg
newDst:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\_Result_\\R.A_1.jpg
newDst2:  _Result Calibration_\Test 1\1_Original-Images\Percobaan 2 - Atap Kamar (Vertikal)\\R.A_1.jpg

oriSrc:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\R.Camera_A_ 05-Jul-2021_16_21_03.jpg
newDst:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\_Result_\\R.A_2.jpg
newDst2:  _Result Calibration_\Test 1\1_Original-Images\Percobaan 2 - Atap Kamar (Vertikal)\\R.A_2.jpg

oriSrc:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\R.Camera_A_ 05-Jul-2021_16_21_38.jpg
newDst:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\_Result_\\R.A_3.jpg
newDst2:  _Result Calibration_\Test 1\1_Original-Images\Percobaan 2 - Atap Kamar (Vertikal)\\R.A_3.jpg

oriSrc:  Percobaan 2 - Atap Kamar (Vertikal)\\Final\\R.Camera_

PermissionError: [Errno 13] Permission denied: 'Percobaan 2 - Atap Kamar (Vertikal)\\\\Final\\\\_Result_'

*** Get Coordinate in New Images ***
==============================

In [2]:
# Get Location in New Images
import cv2
import numpy as np

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderPercobaan = r"Percobaan 2 - Atap Kamar (Vertikal)\\"
# "Percobaan 2 - Atap Kamar (Vertikal)\Final\\"
# "Percobaan 1 - Depan Kamar (Horizontal)\\"

# resultPath = folderPercobaan + r"_Result_\\"
resultPath = folderPercobaan + r"Final\_Result_\\"

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("Folder Percobaan (resultPath): ", resultPath)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

def mousePoints (event, x, y, flags, params):
    global counter
    #print('-event=', event,' -x,y:',x,y,'\t-flags:', flags, ' -params:', params)
    if event == cv2.EVENT_LBUTTONDOWN:
        print(x,y)

        circles[counter] = x,y
        counter = counter + 1
        #print(circles)

counter = 0
circles = np.zeros((4,2), np.int)

# Camera A Best Coordinate
corKriAts = np.zeros((1,2), np.int)
corKanAts = np.zeros((1,2), np.int)
corKanBwh = np.zeros((1,2), np.int)
corKriBwh = np.zeros((1,2), np.int)

# Camera B Best Coordinate (Relate on Camera A)
corKriAtsB = np.zeros((1,2), np.int)
corKanAtsB = np.zeros((1,2), np.int)
corKanBwhB = np.zeros((1,2), np.int)
corKriBwhB = np.zeros((1,2), np.int)
# Camera C Best Coordinate (Relate on Camera A)
corKriAtsC = np.zeros((1,2), np.int)
corKanAtsC = np.zeros((1,2), np.int)
corKanBwhC = np.zeros((1,2), np.int)
corKriBwhC = np.zeros((1,2), np.int)
# Camera D Best Coordinate (Relate on Camera A)
corKriAtsD = np.zeros((1,2), np.int)
corKanAtsD = np.zeros((1,2), np.int)
corKanBwhD = np.zeros((1,2), np.int)
corKriBwhD = np.zeros((1,2), np.int)

# Process Images
#resultPath = r"Percobaan 2 - Atap Kamar (Vertikal)\Final\_Result_\\"
for filename in os.listdir(resultPath):
    print()
    print(filename)
    imagePath = resultPath + filename
    img = cv2.imread(imagePath)

    while True:
        for x in range(0,4):
            cv2.circle(img,(circles[x][0], circles[x][1]), 3, (0,255,0), cv2.FILLED)
        
        cv2.imshow('Original Image ' + str(filename), img)
        cv2.setMouseCallback('Original Image ' + str(filename), mousePoints)    

        if (cv2.waitKey(1) & 0xFF == ord('q')) or counter == 4:
            break
        cv2.waitKey(1)
    cv2.destroyAllWindows()
    counter = 0

    # Find Best Coordinate
    if filename[:3] == Rotate + "A":
        # Camera A

        # Kiri Atas (1)
        simpanX1A = circles[0][0]
        simpanY1A = circles[0][1]
        if simpanX1A <= corKriAts[0][0] or corKriAts[0][0] == 0:
            corKriAts[0][0] = simpanX1A
            # Save Name
            saveNameX1 = filename[4:5]
        if simpanY1A <= corKriAts[0][1] or corKriAts[0][1] == 0:
            corKriAts[0][1] = simpanY1A
            # Save Name
            saveNameY1 = filename[4:5]
        #print('>> x1,y1 = ', corKriAts[0][0], corKriAts[0][1], " , name: ", saveNameX1, saveNameY1)

        # Kanan Atas (2)
        simpanX2A = circles[1][0]
        simpanY2A = circles[1][1]
        if simpanX2A >= corKanAts[0][0] or corKanAts[0][0] == 0:
            corKanAts[0][0] = simpanX2A
            # Save Name
            saveNameX2 = filename[4:5]
        if simpanY2A <= corKanAts[0][1] or corKanAts[0][1] == 0:
            corKanAts[0][1] = simpanY2A
            # Save Name
            saveNameY2 = filename[4:5]
        #print('>> x2,y2 = ', corKanAts[0][0], corKanAts[0][1], " , name: ", saveNameX2, saveNameY2)

        # Kanan Bawah (3)
        simpanX3A = circles[2][0]
        simpanY3A = circles[2][1]
        if simpanX3A >= corKanBwh[0][0] or corKanBwh[0][0] == 0:
            corKanBwh[0][0] = simpanX3A
            # Save Name
            saveNameX3 = filename[4:5]
        if simpanY3A >= corKanBwh[0][1] or corKanBwh[0][1] == 0:
            corKanBwh[0][1] = simpanY3A
            # Save Name
            saveNameY3 = filename[4:5]
        #print('>> x3,y3 = ', corKanBwh[0][0], corKanBwh[0][1], " , name: ", saveNameX3, saveNameY3) 

        # Kiri Bawah (4)
        simpanX4A = circles[3][0]
        simpanY4A = circles[3][1]
        if simpanX4A <= corKriBwh[0][0] or corKriBwh[0][0] == 0:
            corKriBwh[0][0] = simpanX4A
            # Save Name
            saveNameX4 = filename[4:5]
        if simpanY4A >= corKriBwh[0][1] or corKriBwh[0][1] == 0:
            corKriBwh[0][1] = simpanY4A
            # Save Name
            saveNameY4 = filename[4:5]
        #print('>> x4,y4 = ', corKriBwh[0][0], corKriBwh[0][1], " , name: ", saveNameX4, saveNameY4) 

    # Find Best Coordinate (Relate on Coordinate A) "name"
    # Coordinate Camera B
    if filename[:3] == Rotate + "B":
        # Get Coordinate X
        if filename[4:5] == saveNameX1:
            corKriAtsB[0][0] = circles[0][0]
        if filename[4:5] == saveNameX2:
            corKanAtsB[0][0] = circles[1][0]
        if filename[4:5] == saveNameX3:
            corKanBwhB[0][0] = circles[2][0]
        if filename[4:5] == saveNameX4:
            corKriBwhB[0][0] = circles[3][0]

        # Get Coordinate Y
        if filename[4:5] == saveNameY1:
            corKriAtsB[0][1] = circles[0][1]
        if filename[4:5] == saveNameY2:
            corKanAtsB[0][1] = circles[1][1]
        if filename[4:5] == saveNameY3:
            corKanBwhB[0][1] = circles[2][1]
        if filename[4:5] == saveNameY4:
            corKriBwhB[0][1] = circles[3][1]
        
        # Final Coordinate Camera B
        """
        print('>> x1B,y1B = ', corKriAtsB[0][0], corKriAtsB[0][1])
        print('>> x2B,y2B = ', corKanAtsB[0][0], corKanAtsB[0][1])
        print('>> x3B,y3B = ', corKanBwhB[0][0], corKanBwhB[0][1])
        print('>> x4B,y4B = ', corKriBwhB[0][0], corKriBwhB[0][1])
        """

    # Coordinate Camera C
    if filename[:3] == Rotate + "C":
        # Get Coordinate X
        if filename[4:5] == saveNameX1:
            corKriAtsC[0][0] = circles[0][0]
        if filename[4:5] == saveNameX2:
            corKanAtsC[0][0] = circles[1][0]
        if filename[4:5] == saveNameX3:
            corKanBwhC[0][0] = circles[2][0]
        if filename[4:5] == saveNameX4:
            corKriBwhC[0][0] = circles[3][0]

        # Get Coordinate Y
        if filename[4:5] == saveNameY1:
            corKriAtsC[0][1] = circles[0][1]
        if filename[4:5] == saveNameY2:
            corKanAtsC[0][1] = circles[1][1]
        if filename[4:5] == saveNameY3:
            corKanBwhC[0][1] = circles[2][1]
        if filename[4:5] == saveNameY4:
            corKriBwhC[0][1] = circles[3][1]
        
        # Final Coordinate Camera C
        """
        print('>> x1C,y1C = ', corKriAtsC[0][0], corKriAtsC[0][1])
        print('>> x2C,y2C = ', corKanAtsC[0][0], corKanAtsC[0][1])
        print('>> x3C,y3C = ', corKanBwhC[0][0], corKanBwhC[0][1])
        print('>> x4C,y4C = ', corKriBwhC[0][0], corKriBwhC[0][1])
        """

    # Coordinate Camera D
    if filename[:3] == Rotate + "D":
        # Get Coordinate X
        if filename[4:5] == saveNameX1:
            corKriAtsD[0][0] = circles[0][0]
        if filename[4:5] == saveNameX2:
            corKanAtsD[0][0] = circles[1][0]
        if filename[4:5] == saveNameX3:
            corKanBwhD[0][0] = circles[2][0]
        if filename[4:5] == saveNameX4:
            corKriBwhD[0][0] = circles[3][0]

        # Get Coordinate Y
        if filename[4:5] == saveNameY1:
            corKriAtsD[0][1] = circles[0][1]
        if filename[4:5] == saveNameY2:
            corKanAtsD[0][1] = circles[1][1]
        if filename[4:5] == saveNameY3:
            corKanBwhD[0][1] = circles[2][1]
        if filename[4:5] == saveNameY4:
            corKriBwhD[0][1] = circles[3][1]
        
        # Final Coordinate Camera B
        """
        print('>> x1D,y1D = ', corKriAtsD[0][0], corKriAtsD[0][1])
        print('>> x2D,y2D = ', corKanAtsD[0][0], corKanAtsD[0][1])
        print('>> x3D,y3D = ', corKanBwhD[0][0], corKanBwhD[0][1])
        print('>> x4D,y4D = ', corKriBwhD[0][0], corKriBwhD[0][1])
        """

#====================================================================================
# FINAL
print("FINAL Best Coordinate")
print("~"*50)
print()

# Coordinate Kiri Atas
print("Coordinate Kiri Atas")
print("Take from Camera A, X1 = ", saveNameX1, " and Y1 = ", saveNameY1)
print("=-"*25)
print('x1A,y1A = ', corKriAts)
print('x1B,y1B = ', corKriAtsB)
print('x1C,y1C = ', corKriAtsC)
print('x1D,y1D = ', corKriAtsD)
print()

# Coordinate Kanan Atas
print("Coordinate Kanan Atas")
print("Take from Camera A, X2 = ", saveNameX2, " and Y2 = ", saveNameY2)
print("=-"*25)
print('x2A,y2A = ', corKanAts)
print('x2B,y2B = ', corKanAtsB)
print('x2C,y2C = ', corKanAtsC)
print('x2D,y2D = ', corKanAtsD)
print()

# Coordinate Kanan Bawah
print("Coordinate Kanan Bawah")
print("Take from Camera A, X3 = ", saveNameX3, " and Y3 = ", saveNameY3)
print("=-"*25)
print('x3A,y3A = ', corKanBwh)
print('x3B,y3B = ', corKanBwhB)
print('x3C,y3C = ', corKanBwhC)
print('x3D,y3D = ', corKanBwhD)
print()

# Coordinate Kiri Bawah
print("Coordinate Kiri Bawah")
print("Take from Camera A, X4 = ", saveNameX4, " and Y4 = ", saveNameY4)
print("=-"*25)
print('x4A,y4A = ', corKriBwh)
print('x4B,y4B = ', corKriBwhB)
print('x4C,y4C = ', corKriBwhC)
print('x4D,y4D = ', corKriBwhD)
print()


Number Test:  1
Rotate True
Folder Percobaan (resultPath):  Percobaan 2 - Atap Kamar (Vertikal)\\Final\_Result_\\
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>


R.A_1.jpg
59 146
446 140
460 545
62 547

R.A_2.jpg
65 209
467 240
434 635
26 598

R.A_3.jpg
35 88
415 92
411 499
19 482

R.A_4.jpg
69 67
453 58
470 460
74 463

R.A_5.jpg
47 220
432 216
447 626
48 623

R.A_6.jpg
80 126
465 134
463 542
67 522

R.B_1.jpg
51 121
437 114
447 513
64 519

R.B_2.jpg
56 187
460 215
427 601
29 569

R.B_3.jpg
20 61
408 65
403 468
19 456

R.B_4.jpg
57 42
444 26
458 430
75 438

R.B_5.jpg
42 196
425 192
436 587
52 592

R.B_6.jpg
68 98
457 107
451 506
66 496

R.C_1.jpg
39 143
436 125
459 528
65 554

R.C_2.jpg
49 210
458 227
441 620
30 606

R.C_3.jpg
10 82
406 76
412 486
17 491

R.C_4.jpg
44 61
438 39
466 448
74 466

R.C_5.jpg
29 220
428 204
448 609
54 629

R.C_6.jpg
57 117
455 115
464 526
67 529

R.D_1.jpg
42 125
429 112
446 516
61 524

R.D_2.jpg
51 193
451 217
422 603
27 576

R.D_3.jpg
12 68
404 68
400

*** Get Location in New Images [ PRINT FINAL ] ***
===================================================

In [3]:
#====================================================================================
# FINAL
print("FINAL Best Coordinate")
print("~"*50)
print()

# Coordinate Kiri Atas
print("Coordinate Kiri Atas")
print("Take from Camera A, X1 = ", saveNameX1, " and Y1 = ", saveNameY1)
print("=-"*25)
print('x1A,y1A = ', corKriAts)
print('x1B,y1B = ', corKriAtsB)
print('x1C,y1C = ', corKriAtsC)
print('x1D,y1D = ', corKriAtsD)
print()

# Coordinate Kanan Atas
print("Coordinate Kanan Atas")
print("Take from Camera A, X2 = ", saveNameX2, " and Y2 = ", saveNameY2)
print("=-"*25)
print('x2A,y2A = ', corKanAts)
print('x2B,y2B = ', corKanAtsB)
print('x2C,y2C = ', corKanAtsC)
print('x2D,y2D = ', corKanAtsD)
print()

# Coordinate Kanan Bawah
print("Coordinate Kanan Bawah")
print("Take from Camera A, X3 = ", saveNameX3, " and Y3 = ", saveNameY3)
print("=-"*25)
print('x3A,y3A = ', corKanBwh)
print('x3B,y3B = ', corKanBwhB)
print('x3C,y3C = ', corKanBwhC)
print('x3D,y3D = ', corKanBwhD)
print()

# Coordinate Kiri Bawah
print("Coordinate Kiri Bawah")
print("Take from Camera A, X4 = ", saveNameX4, " and Y4 = ", saveNameY4)
print("=-"*25)
print('x4A,y4A = ', corKriBwh)
print('x4B,y4B = ', corKriBwhB)
print('x4C,y4C = ', corKriBwhC)
print('x4D,y4D = ', corKriBwhD)
print()

FINAL Best Coordinate
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Coordinate Kiri Atas
Take from Camera A, X1 =  3  and Y1 =  4
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x1A,y1A =  [[35 67]]
x1B,y1B =  [[20 42]]
x1C,y1C =  [[10 61]]
x1D,y1D =  [[12 43]]

Coordinate Kanan Atas
Take from Camera A, X2 =  2  and Y2 =  4
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x2A,y2A =  [[467  58]]
x2B,y2B =  [[460  26]]
x2C,y2C =  [[458  39]]
x2D,y2D =  [[451  28]]

Coordinate Kanan Bawah
Take from Camera A, X3 =  4  and Y3 =  2
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x3A,y3A =  [[470 635]]
x3B,y3B =  [[458 601]]
x3C,y3C =  [[466 620]]
x3D,y3D =  [[456 603]]

Coordinate Kiri Bawah
Take from Camera A, X4 =  3  and Y4 =  5
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x4A,y4A =  [[ 19 623]]
x4B,y4B =  [[ 19 592]]
x4C,y4C =  [[ 17 629]]
x4D,y4D =  [[ 17 596]]



FINAL Best Coordinate
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Coordinate Kiri Atas
Take from Camera A, X1 =  3  and Y1 =  4
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x1A,y1A =  [[33 65]]
x1B,y1B =  [[19 38]]
x1C,y1C =  [[ 7 58]]
x1D,y1D =  [[ 9 45]]

Coordinate Kanan Atas
Take from Camera A, X2 =  2  and Y2 =  4
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x2A,y2A =  [[471  52]]
x2B,y2B =  [[461  24]]
x2C,y2C =  [[461  35]]
x2D,y2D =  [[455  24]]

Coordinate Kanan Bawah
Take from Camera A, X3 =  4  and Y3 =  2
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x3A,y3A =  [[470 638]]
x3B,y3B =  [[462 602]]
x3C,y3C =  [[470 622]]
x3D,y3D =  [[458 603]]

Coordinate Kiri Bawah
Take from Camera A, X4 =  3  and Y4 =  5
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x4A,y4A =  [[ 18 631]]
x4B,y4B =  [[ 16 597]]
x4C,y4C =  [[ 16 633]]
x4D,y4D =  [[ 14 599]]

*** Get Location in New Images [ SETTING MANUAL ] ***
===================================================

In [6]:
#====================================================================================
# FINAL
print("FINAL Best Coordinate")
print("~"*50)
print()

saveNameX1 = 3
saveNameX2 = 2
saveNameX3 = 4
saveNameX4 = 3
#======================
saveNameY1 = 4
saveNameY2 = 4
saveNameY3 = 2
saveNameY4 = 5

# Coordinate Kiri Atas
#==============================
corKriAts = [[33, 65]]
corKriAtsB = [[19, 38]]
corKriAtsC = [[ 7, 58]]
corKriAtsD = [[ 9, 45]]
#==============================
print("Coordinate Kiri Atas")
print("Take from Camera A, X1 = ", saveNameX1, " and Y1 = ", saveNameY1)
print("=-"*25)
print('x1A,y1A = ', corKriAts)
print('x1B,y1B = ', corKriAtsB)
print('x1C,y1C = ', corKriAtsC)
print('x1D,y1D = ', corKriAtsD)
print()

# Coordinate Kanan Atas
#==============================
corKanAts = [[471,  52]]
corKanAtsB = [[461,  24]]
corKanAtsC = [[461,  35]]
corKanAtsD = [[455,  24]]
#==============================
print("Coordinate Kanan Atas")
print("Take from Camera A, X2 = ", saveNameX2, " and Y2 = ", saveNameY2)
print("=-"*25)
print('x2A,y2A = ', corKanAts)
print('x2B,y2B = ', corKanAtsB)
print('x2C,y2C = ', corKanAtsC)
print('x2D,y2D = ', corKanAtsD)
print()

# Coordinate Kanan Bawah
#==============================
corKanBwh = [[470, 638]]
corKanBwhB = [[462, 602]]
corKanBwhC = [[470, 622]]
corKanBwhD = [[458, 603]]
#==============================
print("Coordinate Kanan Bawah")
print("Take from Camera A, X3 = ", saveNameX3, " and Y3 = ", saveNameY3)
print("=-"*25)
print('x3A,y3A = ', corKanBwh)
print('x3B,y3B = ', corKanBwhB)
print('x3C,y3C = ', corKanBwhC)
print('x3D,y3D = ', corKanBwhD)
print()

# Coordinate Kiri Bawah
#==============================
corKriBwh = [[ 18, 631]]
corKriBwhB = [[ 16, 597]]
corKriBwhC = [[ 16, 633]]
corKriBwhD = [[ 14, 599]]
#==============================
print("Coordinate Kiri Bawah")
print("Take from Camera A, X4 = ", saveNameX4, " and Y4 = ", saveNameY4)
print("=-"*25)
print('x4A,y4A = ', corKriBwh)
print('x4B,y4B = ', corKriBwhB)
print('x4C,y4C = ', corKriBwhC)
print('x4D,y4D = ', corKriBwhD)
print()

FINAL Best Coordinate
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Coordinate Kiri Atas
Take from Camera A, X1 =  3  and Y1 =  4
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x1A,y1A =  [[33, 65]]
x1B,y1B =  [[19, 38]]
x1C,y1C =  [[7, 58]]
x1D,y1D =  [[9, 45]]

Coordinate Kanan Atas
Take from Camera A, X2 =  2  and Y2 =  4
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x2A,y2A =  [[471, 52]]
x2B,y2B =  [[461, 24]]
x2C,y2C =  [[461, 35]]
x2D,y2D =  [[455, 24]]

Coordinate Kanan Bawah
Take from Camera A, X3 =  4  and Y3 =  2
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x3A,y3A =  [[470, 638]]
x3B,y3B =  [[462, 602]]
x3C,y3C =  [[470, 622]]
x3D,y3D =  [[458, 603]]

Coordinate Kiri Bawah
Take from Camera A, X4 =  3  and Y4 =  5
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
x4A,y4A =  [[18, 631]]
x4B,y4B =  [[16, 597]]
x4C,y4C =  [[16, 633]]
x4D,y4D =  [[14, 599]]



*** Draw Line to Get Perspective ***
===================================================

In [4]:
import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderDrawLine = r"_Result Calibration_\Test " + str(numberTest) + "\\2.0_Line Callibration\\"

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("Folder folderDrawLine: ", folderDrawLine)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
# Draw Line
colorA = (0,0,255)
colorB = (0,255,0)
colorC = (255,0,0)
colorD = (0,255,255)
Thickness=2


# Camera A
blankA = np.zeros((640,480,3), dtype='uint8')
cv2.imshow('Blank', blankA)
# 4. Draw a line
cv2.line(blankA, tuple(corKriAts[0]), tuple(corKanAts[0]), colorA, Thickness)
cv2.imshow('Blank', blankA)
cv2.line(blankA, tuple(corKanAts[0]), tuple(corKanBwh[0]), colorA, Thickness)
cv2.imshow('Blank', blankA)
cv2.line(blankA, tuple(corKanBwh[0]), tuple(corKriBwh[0]), colorA, Thickness)
cv2.imshow('Blank', blankA)
cv2.destroyWindow('Blank')
cv2.line(blankA, tuple(corKriBwh[0]), tuple(corKriAts[0]), colorA, Thickness)
cv2.imshow('Camera A', blankA)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera A.jpg", blankA)

# Camera B
blankB = np.zeros((640,480,3), dtype='uint8')
cv2.imshow('Blank', blankB)
# 4. Draw a line
cv2.line(blankB, tuple(corKriAtsB[0]), tuple(corKanAtsB[0]), colorB, Thickness)
cv2.imshow('Blank', blankB)
cv2.line(blankB, tuple(corKanAtsB[0]), tuple(corKanBwhB[0]), colorB, Thickness)
cv2.imshow('Blank', blankB)
cv2.line(blankB, tuple(corKanBwhB[0]), tuple(corKriBwhB[0]), colorB, Thickness)
cv2.imshow('Blank', blankB)
cv2.destroyWindow('Blank')
cv2.line(blankB, tuple(corKriBwhB[0]), tuple(corKriAtsB[0]), colorB, Thickness)
cv2.imshow('Camera B', blankB)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera B.jpg", blankB)

# Camera C
blankC = np.zeros((640,480,3), dtype='uint8')
cv2.imshow('Blank', blankC)
# 4. Draw a line
cv2.line(blankC, tuple(corKriAtsC[0]), tuple(corKanAtsC[0]), colorC, Thickness)
cv2.imshow('Blank', blankC)
cv2.line(blankC, tuple(corKanAtsC[0]), tuple(corKanBwhC[0]), colorC, Thickness)
cv2.imshow('Blank', blankC)
cv2.line(blankC, tuple(corKanBwhC[0]), tuple(corKriBwhC[0]), colorC, Thickness)
cv2.imshow('Blank', blankC)
cv2.destroyWindow('Blank')
cv2.line(blankC, tuple(corKriBwhC[0]), tuple(corKriAtsC[0]), colorC, Thickness)
cv2.imshow('Camera C', blankC)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera C.jpg", blankC)

# Camera D
blankD = np.zeros((640,480,3), dtype='uint8')
cv2.imshow('Blank', blankD)
# 4. Draw a line
cv2.line(blankD, tuple(corKriAtsD[0]), tuple(corKanAtsD[0]), colorD, Thickness)
cv2.imshow('Blank', blankD)
cv2.line(blankD, tuple(corKanAtsD[0]), tuple(corKanBwhD[0]), colorD, Thickness)
cv2.imshow('Blank', blankD)
cv2.line(blankD, tuple(corKanBwhD[0]), tuple(corKriBwhD[0]), colorD, Thickness)
cv2.imshow('Blank', blankD)
cv2.destroyWindow('Blank')
cv2.line(blankD, tuple(corKriBwhD[0]), tuple(corKriAtsD[0]), colorD, Thickness)
cv2.imshow('Camera D', blankD)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera D.jpg", blankD)

cv2.waitKey(0)
cv2.destroyAllWindows()

Number Test:  1
Rotate True
Folder folderDrawLine:  _Result Calibration_\Test 1\2.0_Line Callibration\
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>



*** Get Best Pixel ***
===================================================

In [5]:
# Get Best Pixel

import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderDrawLine = r"_Result Calibration_\Test " + str(numberTest) + "\\2.0_Line Callibration\\"

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("Folder folderDrawLine: ", folderDrawLine)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# Camera A
print("Camera A Pixel Rectangle")
print("="*50)
#== Shortest X
if corKriAts[0][0] <= corKriBwh[0][0]:
    xMinA = corKriAts[0][0]
else:
    xMinA = corKriBwh[0][0]
#==  Shortest Y
if corKriAts[0][1] <= corKanAts[0][1]:
    yMinA = corKriAts[0][1]
else:
    yMinA = corKanAts[0][1]
#====== Longest X
if corKanAts[0][0] >= corKanBwh[0][0]:
    xMaxA = corKanAts[0][0]
else:
    xMaxA = corKanBwh[0][0]
#====== Longest Y
if corKriBwh[0][1] >= corKanBwh[0][1]:
    yMaxA = corKriBwh[0][1]
else:
    yMaxA = corKanBwh[0][1]
pixelXA = xMaxA - xMinA
pixelYA = yMaxA - yMinA
print("Coordinate Shortest A = (", xMinA, yMinA, ") and Longest = (", xMaxA, yMaxA, ")")
print("Pixel = ", pixelXA, "x", pixelYA)
print()

# Camera B
print("Camera B Pixel Rectangle")
print("="*50)
#== Shortest X
if corKriAtsB[0][0] <= corKriBwhB[0][0]:
    xMinB = corKriAtsB[0][0]
else:
    xMinB = corKriBwhB[0][0]
#==  Shortest Y
if corKriAtsB[0][1] <= corKanAtsB[0][1]:
    yMinB = corKriAtsB[0][1]
else:
    yMinB = corKanAtsB[0][1]
#====== Longest X
if corKanAtsB[0][0] >= corKanBwhB[0][0]:
    xMaxB = corKanAtsB[0][0]
else:
    xMaxB = corKanBwhB[0][0]
#====== Longest Y
if corKriBwhB[0][1] >= corKanBwhB[0][1]:
    yMaxB = corKriBwhB[0][1]
else:
    yMaxB = corKanBwhB[0][1]
pixelXB = xMaxB - xMinB
pixelYB = yMaxB - yMinB
print("Coordinate Shortest B = (", xMinB, yMinB, ") and Longest = (", xMaxB, yMaxB, ")")
print("Pixel = ", pixelXB, "x", pixelYB)
print()

# Camera C
print("Camera C Pixel Rectangle")
print("="*50)
#== Shortest X
if corKriAtsC[0][0] <= corKriBwhC[0][0]:
    xMinC = corKriAtsC[0][0]
else:
    xMinC = corKriBwhC[0][0]
#==  Shortest Y
if corKriAtsC[0][1] <= corKanAtsC[0][1]:
    yMinC = corKriAtsC[0][1]
else:
    yMinC = corKanAtsC[0][1]
#====== Longest X
if corKanAtsC[0][0] >= corKanBwhC[0][0]:
    xMaxC = corKanAtsC[0][0]
else:
    xMaxC = corKanBwhC[0][0]
#====== Longest Y
if corKriBwhC[0][1] >= corKanBwhC[0][1]:
    yMaxC = corKriBwhC[0][1]
else:
    yMaxC = corKanBwhC[0][1]
pixelXC = xMaxC - xMinC
pixelYC = yMaxC - yMinC
print("Coordinate Shortest C = (", xMinC, yMinC, ") and Longest = (", xMaxC, yMaxC, ")")
print("Pixel = ", pixelXC, "x", pixelYC)
print()

# Camera D
print("Camera D Pixel Rectangle")
print("="*50)
#== Shortest X
if corKriAtsD[0][0] <= corKriBwhD[0][0]:
    xMinD = corKriAtsD[0][0]
else:
    xMinD = corKriBwhD[0][0]
#==  Shortest Y
if corKriAtsD[0][1] <= corKanAtsD[0][1]:
    yMinD = corKriAtsD[0][1]
else:
    yMinD = corKanAtsD[0][1]
#====== Longest X
if corKanAtsD[0][0] >= corKanBwhD[0][0]:
    xMaxD = corKanAtsD[0][0]
else:
    xMaxD = corKanBwhD[0][0]
#====== Longest Y
if corKriBwhD[0][1] >= corKanBwhD[0][1]:
    yMaxD = corKriBwhD[0][1]
else:
    yMaxD = corKanBwhD[0][1]
pixelXD = xMaxD - xMinD
pixelYD = yMaxD - yMinD
print("Coordinate Shortest D = (", xMinD, yMinD, ") and Longest = (", xMaxD, yMaxD, ")")
print("Pixel = ", pixelXD, "x", pixelYD)
print()

# Write Rectangle
lineA = cv2.imread(folderDrawLine + "R.Line Perspective - Camera A.jpg")
lineB = cv2.imread(folderDrawLine + "R.Line Perspective - Camera B.jpg")
lineC = cv2.imread(folderDrawLine + "R.Line Perspective - Camera C.jpg")
lineD = cv2.imread(folderDrawLine + "R.Line Perspective - Camera D.jpg")
colorRectangle = (255, 255, 255)
ThicknessRect = 1

# Rectangle A
cv2.imshow('Rectangle A', lineA)
cv2.rectangle(lineA, (xMinA, yMinA), (xMaxA, yMaxA), colorRectangle, ThicknessRect)
cv2.imshow('Rectangle A', lineA)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera A (Rectangle).jpg", lineA)
# Rectangle B
cv2.imshow('Rectangle B', lineB)
cv2.rectangle(lineB, (xMinB, yMinB), (xMaxB, yMaxB), colorRectangle, ThicknessRect)
cv2.imshow('Rectangle B', lineB)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera B (Rectangle).jpg", lineB)
# Rectangle C
cv2.imshow('Rectangle C', lineC)
cv2.rectangle(lineC, (xMinC, yMinC), (xMaxC, yMaxC), colorRectangle, ThicknessRect)
cv2.imshow('Rectangle C', lineC)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera C (Rectangle).jpg", lineC)
# Rectangle D
cv2.imshow('Rectangle D', lineD)
cv2.rectangle(lineD, (xMinD, yMinD), (xMaxD, yMaxD), colorRectangle, ThicknessRect)
cv2.imshow('Rectangle D', lineD)
cv2.imwrite(folderDrawLine + Rotate + "Line Perspective - Camera D (Rectangle).jpg", lineD)

cv2.waitKey(0)
cv2.destroyAllWindows()

Number Test:  1
Rotate True
Folder folderDrawLine:  _Result Calibration_\Test 1\2.0_Line Callibration\
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

Camera A Pixel Rectangle
Coordinate Shortest A = ( 19 58 ) and Longest = ( 470 635 )
Pixel =  451 x 577

Camera B Pixel Rectangle
Coordinate Shortest B = ( 19 26 ) and Longest = ( 460 601 )
Pixel =  441 x 575

Camera C Pixel Rectangle
Coordinate Shortest C = ( 10 39 ) and Longest = ( 466 629 )
Pixel =  456 x 590

Camera D Pixel Rectangle
Coordinate Shortest D = ( 12 28 ) and Longest = ( 456 603 )
Pixel =  444 x 575



*** Get Best Pixel [ PRINT FINAL ] ***
===================================================
*** Camera A Pixel Rectangle ***
Coordinate Shortest A = ( 18 52 ) and Longest = ( 471 638 )
Pixel =  453 x 586

*** Camera B Pixel Rectangle ***
Coordinate Shortest B = ( 16 24 ) and Longest = ( 462 602 )
Pixel =  446 x 578

*** Camera C Pixel Rectangle ***
Coordinate Shortest C = ( 7 35 ) and Longest = ( 470 633 )
Pixel =  463 x 598

*** Camera D Pixel Rectangle ***
Coordinate Shortest D = ( 9 24 ) and Longest = ( 458 603 )
Pixel =  449 x 579

*** ============================================================= ***
=========================================================================

*** Warp Perpsective ***
===================================================

In [6]:
import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderCallibration = r"_Result Calibration_\\Test " + str(numberTest) + r"\\2_Calibration-Images\\"
folderOriginalImages = r"_Result Calibration_\\Test " + str(numberTest) + r"\\1_Original-Images\\"
folderImages = r"Percobaan 2 - Atap Kamar (Vertikal)\\"

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("folderCallibration: ", folderCallibration)
print("folderOriginalImages: ", folderOriginalImages)
print("folderImages: ", folderImages)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# WARP Perspective
noImage = 1
print("=> no. Image = ", noImage)
imgA = cv2.imread(folderOriginalImages + folderImages + Rotate + "A_" + str(noImage) + ".jpg")

cv2.imshow('Camera A - Perspective', imgA)

# CAMERA A - B
imgB = cv2.imread(folderOriginalImages + folderImages + Rotate + "B_" + str(noImage) + ".jpg")
cv2.imshow('Camera B - Perspective', imgB)
# Coordinate A
pts1 = np.float32([corKriAts[0],corKanAts[0],corKriBwh[0],corKanBwh[0]])
# Coordinate B
pts2 = np.float32([corKriAtsB[0],corKanAtsB[0],corKriBwhB[0],corKanBwhB[0]])
matrix = cv2.getPerspectiveTransform(pts2,pts1)
imgOutputB = cv2.warpPerspective(imgB,matrix,(480,640))

# CAMERA A - C
imgC = cv2.imread(folderOriginalImages + folderImages + Rotate + "C_" + str(noImage) + ".jpg")
# Coordinate A
pts1 = np.float32([corKriAts[0],corKanAts[0],corKriBwh[0],corKanBwh[0]])
# Coordinate C
pts2 = np.float32([corKriAtsC[0],corKanAtsC[0],corKriBwhC[0],corKanBwhC[0]])
matrix = cv2.getPerspectiveTransform(pts2,pts1)
imgOutputC = cv2.warpPerspective(imgC,matrix,(480,640))

# CAMERA A - D
imgD = cv2.imread(folderOriginalImages + folderImages + Rotate + "D_" + str(noImage) + ".jpg")
# Coordinate A
pts1 = np.float32([corKriAts[0],corKanAts[0],corKriBwh[0],corKanBwh[0]])
# Coordinate D
pts2 = np.float32([corKriAtsD[0],corKanAtsD[0],corKriBwhD[0],corKanBwhD[0]])
matrix = cv2.getPerspectiveTransform(pts2,pts1)
imgOutputD = cv2.warpPerspective(imgD,matrix,(480,640))

cv2.imshow('Camera A - Perspective', imgA)
cv2.imshow('Camera B - Perspective', imgOutputB)
cv2.imshow('Camera C - Perspective', imgOutputC)
cv2.imshow('Camera D - Perspective', imgOutputD)
#==== W R I T E
anotherParams = 1
cv2.imwrite(folderCallibration + Rotate + str(noImage) + "." + str(anotherParams) + "-A.jpg", imgA)
cv2.imwrite(folderCallibration + Rotate + str(noImage) + "." + str(anotherParams) + "-B.jpg", imgOutputB)
cv2.imwrite(folderCallibration + Rotate + str(noImage) + "." + str(anotherParams) + "-C.jpg", imgOutputC)
cv2.imwrite(folderCallibration + Rotate + str(noImage) + "." + str(anotherParams) + "-D.jpg", imgOutputD)

cv2.waitKey(0)
cv2.destroyAllWindows()

Number Test:  1
Rotate True
folderCallibration:  _Result Calibration_\\Test 1\\2_Calibration-Images\\
folderOriginalImages:  _Result Calibration_\\Test 1\\1_Original-Images\\
folderImages:  Percobaan 2 - Atap Kamar (Vertikal)\\
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

=> no. Image =  1


*** Get Pixel to Crop ***
=========================

In [7]:
# Get Location in New Images
import cv2
import numpy as np

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderCallibration = r"_Result Calibration_\\Test " + str(numberTest) + r"\\2_Calibration-Images\\"
noImage = 1
anotherParams = 1
imagesName = Rotate + str(noImage) + "." + str(anotherParams)

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("folderCallibration: ", folderCallibration)
print("noImage: ", noImage)
print("anotherParams: ", anotherParams)
print("imagesName: ", imagesName)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

def mousePoints (event, x, y, flags, params):
    global counter
    global circles
    #print('-event=', event,' -x,y:',x,y,'\t-flags:', flags, ' -params:', params)
    if event == cv2.EVENT_LBUTTONDOWN:
        print(x,y)

        circles[counter] = x,y
        counter = counter + 1
        #print(circles)

counter = 0
circles = np.zeros((4,2), np.int)

# Camera A Best Coordinate
corKriAtsCrop = np.zeros((2), np.int)
corKanAtsCrop = np.zeros((2), np.int)
corKanBwhCrop = np.zeros((2), np.int)
corKriBwhCrop = np.zeros((2), np.int)

# Camera B Best Coordinate (Relate on Camera A)
corKriAtsCropB = np.zeros((2), np.int)
corKanAtsCropB = np.zeros((2), np.int)
corKanBwhCropB = np.zeros((2), np.int)
corKriBwhCropB = np.zeros((2), np.int)
# Camera C Best Coordinate (Relate on Camera A)
corKriAtsCropC = np.zeros((2), np.int)
corKanAtsCropC = np.zeros((2), np.int)
corKanBwhCropC = np.zeros((2), np.int)
corKriBwhCropC = np.zeros((2), np.int)
# Camera D Best Coordinate (Relate on Camera A)
corKriAtsCropD = np.zeros((2), np.int)
corKanAtsCropD = np.zeros((2), np.int)
corKanBwhCropD = np.zeros((2), np.int)
corKriBwhCropD = np.zeros((2), np.int)

# Process Images
# resultPath = r"_Result Calibration_\Test 1\\"
#imagesName = R + str(noImage) + "." + str(anotherParams)
for filename in os.listdir(folderCallibration):
    if filename[:5] == imagesName:
        print(filename)
        imagePath = folderCallibration + filename
        img = cv2.imread(imagePath)

        while True:
            for x in range(0,4):
                cv2.circle(img,(circles[x][0], circles[x][1]), 3, (0,255,0), cv2.FILLED)
            
            cv2.imshow('Original Image ' + str(filename), img)
            cv2.setMouseCallback('Original Image ' + str(filename), mousePoints)    

            if (cv2.waitKey(1) & 0xFF == ord('q')) or counter == 4:
                break
            cv2.waitKey(1)

        # Get Coordinate Camera B
        if filename[5:7] == "-B":
            corKriAtsCropB = circles[[0]]
            corKanAtsCropB = circles[[1]]
            corKanBwhCropB = circles[[2]]
            corKriBwhCropB = circles[[3]]

        # Get Coordinate Camera C
        if filename[5:7] == "-C":
            corKriAtsCropC = circles[[0]]
            corKanAtsCropC = circles[[1]]
            corKanBwhCropC = circles[[2]]
            corKriBwhCropC = circles[[3]]

        # Get Coordinate Camera D
        if filename[5:7] == "-D":
            corKriAtsCropD = circles[[0]]
            corKanAtsCropD = circles[[1]]
            corKanBwhCropD = circles[[2]]
            corKriBwhCropD = circles[[3]]
    cv2.destroyAllWindows()
    counter = 0

print()
# Crop Coordinate
CropKriAts = np.zeros((2), np.int)
CropKanAts = np.zeros((2), np.int)
CropKanBwh = np.zeros((2), np.int)
CropKriBwh = np.zeros((2), np.int)

# ============== Optimal Crop Kiri Atas (X++, Y++)
print(corKriAtsCropB, corKriAtsCropC, corKriAtsCropD)
for i in range (0,1+1):
    # i=0 => X Coordinate, i=1 => Y Coordinate
    if corKriAtsCropB[0][i] >= corKriAtsCropC[0][i] and corKriAtsCropB[0][i] >= corKriAtsCropD[0][i]:
        CropKriAts[i] = corKriAtsCropB[0][i]
    elif corKriAtsCropC[0][i] >= corKriAtsCropB[0][i] and corKriAtsCropC[0][i] >= corKriAtsCropD[0][i]:
        CropKriAts[i] = corKriAtsCropC[0][i]
    elif corKriAtsCropD[0][i] >= corKriAtsCropB[0][i] and corKriAtsCropD[0][i] >= corKriAtsCropC[0][i]:
        CropKriAts[i] = corKriAtsCropD[0][i]
print("CropKriAts: ", CropKriAts)
print()

# ============== Optimal Crop Kanan Atas (X--, Y++)
print(corKanAtsCropB, corKanAtsCropC, corKanAtsCropD)
for i in range (0,1+1):
    # i=0 => X Coordinate, i=1 => Y Coordinate
    if i == 0:
        if corKanAtsCropB[0][i] <= corKanAtsCropC[0][i] and corKanAtsCropB[0][i] <= corKanAtsCropD[0][i]:
            CropKanAts[i] = corKanAtsCropB[0][i]
        elif corKanAtsCropC[0][i] <= corKanAtsCropB[0][i] and corKanAtsCropC[0][i] <= corKanAtsCropD[0][i]:
            CropKanAts[i] = corKanAtsCropC[0][i]
        elif corKanAtsCropD[0][i] <= corKanAtsCropB[0][i] and corKanAtsCropD[0][i] <= corKanAtsCropC[0][i]:
            CropKanAts[i] = corKanAtsCropD[0][i]
    else:
        if corKanAtsCropB[0][i] >= corKanAtsCropC[0][i] and corKanAtsCropB[0][i] >= corKanAtsCropD[0][i]:
            CropKanAts[i] = corKanAtsCropB[0][i]
        elif corKanAtsCropC[0][i] >= corKanAtsCropB[0][i] and corKanAtsCropC[0][i] >= corKanAtsCropD[0][i]:
            CropKanAts[i] = corKanAtsCropC[0][i]
        elif corKanAtsCropD[0][i] >= corKanAtsCropB[0][i] and corKanAtsCropD[0][i] >= corKanAtsCropC[0][i]:
            CropKanAts[i] = corKanAtsCropD[0][i]
print("CropKanAts: ", CropKanAts)
print()

# ============== Optimal Crop Kanan Bawah (X--, Y--)
print(corKanBwhCropB, corKanBwhCropC, corKanBwhCropD)
for i in range (0,1+1):
    # i=0 => X Coordinate, i=1 => Y Coordinate
    if corKanBwhCropB[0][i] <= corKanBwhCropC[0][i] and corKanBwhCropB[0][i] <= corKanBwhCropD[0][i]:
        CropKanBwh[i] = corKanBwhCropB[0][i]
    elif corKanBwhCropC[0][i] <= corKanBwhCropB[0][i] and corKanBwhCropC[0][i] <= corKanBwhCropD[0][i]:
        CropKanBwh[i] = corKanBwhCropC[0][i]
    elif corKanBwhCropD[0][i] <= corKanBwhCropB[0][i] and corKanBwhCropD[0][i] <= corKanBwhCropC[0][i]:
        CropKanBwh[i] = corKanBwhCropD[0][i]
print("CropKanBwh: ", CropKanBwh)
print()

# ============== Optimal Crop Kiri Bawah (X++, Y--)
print(corKriBwhCropB, corKriBwhCropC, corKriBwhCropD)
for i in range (0,1+1):
    # i=0 => X Coordinate, i=1 => Y Coordinate
    if i == 1:
        if corKriBwhCropB[0][i] <= corKriBwhCropC[0][i] and corKriBwhCropB[0][i] <= corKriBwhCropD[0][i]:
            CropKriBwh[i] = corKriBwhCropB[0][i]
        elif corKriBwhCropC[0][i]<= corKriBwhCropB[0][i] and corKriBwhCropC[0][i] <= corKriBwhCropD[0][i]:
            CropKriBwh[i] = corKriBwhCropC[0][i]
        elif corKriBwhCropD[0][i] <= corKriBwhCropB[0][i] and corKriBwhCropD[0][i] <= corKriBwhCropC[0][i]:
            CropKriBwh[i] = corKriBwhCropD[0][i]
    else:
        if corKriBwhCropB[0][i] >= corKriBwhCropC[0][i] and corKriBwhCropB[0][i] >= corKriBwhCropD[0][i]:
            CropKriBwh[i] = corKriBwhCropB[0][i]
        elif corKriBwhCropC[0][i] >= corKriBwhCropB[0][i] and corKriBwhCropC[0][i] >= corKriBwhCropD[0][i]:
            CropKriBwh[i] = corKriBwhCropC[0][i]
        elif corKriBwhCropD[0][i] >= corKriBwhCropB[0][i] and corKriBwhCropD[0][i] >= corKriBwhCropC[0][i]:
            CropKriBwh[i] = corKriBwhCropD[0][i]
print("CropKriBwh: ", CropKriBwh)
print()


Number Test:  1
Rotate True
folderCallibration:  _Result Calibration_\\Test 1\\2_Calibration-Images\\
noImage:  1
anotherParams:  1
imagesName:  R.1.1
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

R.1.1-A.jpg
57 138
445 131
462 549
60 553
R.1.1-B.jpg
19 29
476 37
478 638
1 637
R.1.1-C.jpg
30 13
477 25
475 636
5 627
R.1.1-D.jpg
28 31
476 35
476 636
4 634

[[19 29]] [[30 13]] [[28 31]]
CropKriAts:  [30 31]

[[476  37]] [[477  25]] [[476  35]]
CropKanAts:  [476  37]

[[478 638]] [[475 636]] [[476 636]]
CropKanBwh:  [475 636]

[[  1 637]] [[  5 627]] [[  4 634]]
CropKriBwh:  [  5 627]



*** Get Pixel to Crop [ SETTING MANUAL ] ***
=============================================

In [8]:
# Crop Coordinate
CropKriAts = np.zeros((2), np.int)
CropKanAts = np.zeros((2), np.int)
CropKanBwh = np.zeros((2), np.int)
CropKriBwh = np.zeros((2), np.int)

CropKriAts = [33, 32]
CropKanAts = [475, 33]
CropKanBwh = [476, 634]
CropKriBwh = [7, 631]

print("CropKriAts: ", CropKriAts)
print("CropKanAts: ", CropKanAts)
print("CropKanBwh: ", CropKanBwh)
print("CropKriBwh: ", CropKriBwh)

CropKriAts:  [33, 32]
CropKanAts:  [475, 33]
CropKanBwh:  [476, 634]
CropKriBwh:  [7, 631]


Test1.1-A.jpg
Test1.1-B.jpg
20 32
476 33
476 637
4 636
Test1.1-C.jpg
33 15
475 24
476 634
6 631
Test1.1-D.jpg
29 25
477 33
476 637
7 636
Test1.1-Ori-B.jpg
Test1.1-Ori-C.jpg
Test1.1-Ori-D.jpg

[[20 32]] [[33 15]] [[29 25]]
CropKriAts:  [33 32]

[[476  33]] [[475  24]] [[477  33]]
CropKanAts:  [475  33]

[[476 637]] [[476 634]] [[476 637]]
CropKanBwh:  [476 634]

[[  4 636]] [[  6 631]] [[  7 636]]
CropKriBwh:  [  7 631]

*** Get Best Pixel to Crop Images ***
===================================================

In [9]:
# Get Best Pixel to Crop Images

print("Get Best Pixel to Crop Images")
print("="*50)
# Start (Kiri Atas)
#== Longest X
if CropKriAts[0] >= CropKriBwh[0]:
    xMinCrop = CropKriAts[0]
else:
    xMinCrop = CropKriBwh[0]
#==  Longest Y
if CropKriAts[1] >= CropKanAts[1]:
    yMinCrop = CropKriAts[1]
else:
    yMinCrop = CropKanAts[1]

# End (Kanan Bawah)
#====== Short X
if CropKanAts[0] <= CropKanBwh[0]:
    xMaxCrop = CropKanAts[0]
else:
    xMaxCrop = CropKanBwh[0]
#====== Short Y
if CropKriBwh[1] <= CropKanBwh[1]:
    yMaxCrop = CropKriBwh[1]
else:
    yMaxCrop = CropKanBwh[1]
pixelXCrop = xMaxCrop - xMinCrop
pixelYCrop = yMaxCrop - yMinCrop
print("Coordinate Start = (", xMinCrop, yMinCrop, ") and End = (", xMaxCrop, yMaxCrop, ")")
print("Pixel = ", pixelXCrop, "x", pixelYCrop)
print()

Get Best Pixel to Crop Images
Coordinate Start = ( 33 33 ) and End = ( 475 631 )
Pixel =  442 x 598



*** Draw Rectangle to Crop Images ***
===================================================

In [10]:
import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderCallibration = r"_Result Calibration_\\Test " + str(numberTest) + r"\\2_Calibration-Images\\"
folderRectangleCrop = r"_Result Calibration_\\Test " + str(numberTest) + r"\\3.0_Rectangle Crop\\"

noImage = 1
anotherParams = 1
imagesName = Rotate + str(noImage) + "." + str(anotherParams)

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("folderCallibration: ", folderCallibration)
print("folderRectangleCrop: ", folderRectangleCrop)
print("noImage: ", noImage)
print("anotherParams: ", anotherParams)
print("imagesName: ", imagesName)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# Draw Rectangle
start_point = (xMinCrop, yMinCrop)
end_point = (xMaxCrop, yMaxCrop)
color = (0, 255, 0)
thickness = 2

# Process Images
for filename in os.listdir(folderCallibration):
    if filename[:5] == imagesName:
        print(filename)
        imagePath = folderCallibration + filename
        img = cv2.imread(imagePath)
        # Rectangle
        img = cv2.rectangle(img, start_point, end_point, color, thickness)
        cv2.imshow('Original Image ' + str(filename), img)

        cv2.imwrite(folderRectangleCrop + filename[:7] + "-Rectangle.jpg", img)
        
cv2.waitKey(0)
cv2.destroyAllWindows()

Number Test:  1
Rotate True
folderCallibration:  _Result Calibration_\\Test 1\\2_Calibration-Images\\
folderRectangleCrop:  _Result Calibration_\\Test 1\\3.0_Rectangle Crop\\
noImage:  1
anotherParams:  1
imagesName:  R.1.1
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

R.1.1-A.jpg
R.1.1-B.jpg
R.1.1-C.jpg
R.1.1-D.jpg


*** Best Crop Images ***
===================================================

In [11]:
# Crop

import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderCallibration = r"_Result Calibration_\\Test " + str(numberTest) + r"\\2_Calibration-Images\\"
folderCropImages = r"_Result Calibration_\\Test " + str(numberTest) + r"\\3_Crop-Images (Final)\\"

noImage = 1
anotherParams = 1
imagesName = Rotate + str(noImage) + "." + str(anotherParams)

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("folderCallibration: ", folderCallibration)
print("folderCropImages: ", folderCropImages)
print("noImage: ", noImage)
print("anotherParams: ", anotherParams)
print("imagesName: ", imagesName)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# Crop Rectangle
start_point = (xMinCrop, yMinCrop)
end_point = (xMaxCrop, yMaxCrop)

# Process Images
for filename in os.listdir(folderCallibration):
    if filename[:5] == imagesName:
        print(filename)
        imagePath = folderCallibration + filename
        img = cv2.imread(imagePath)
        img = img[yMinCrop:yMaxCrop+1, xMinCrop:xMaxCrop+1]
        cv2.imshow('Original Image ' + str(filename), img)
        # Crop

        cv2.imwrite(folderCropImages + filename[:7] + "-Final.jpg", img)
        
cv2.waitKey(0)
cv2.destroyAllWindows()

Number Test:  1
Rotate True
folderCallibration:  _Result Calibration_\\Test 1\\2_Calibration-Images\\
folderCropImages:  _Result Calibration_\\Test 1\\3_Crop-Images (Final)\\
noImage:  1
anotherParams:  1
imagesName:  R.1.1
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

R.1.1-A.jpg
R.1.1-B.jpg
R.1.1-C.jpg
R.1.1-D.jpg


*** Test Images - Substraction (grayscale)***
===================================================

In [12]:
# Substraction grayscale

import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderCropImages = r"_Result Calibration_\\Test " + str(numberTest) + r"\\3_Crop-Images (Final)\\"
folderSubstractionImages = r"_Result Calibration_\\Test " + str(numberTest) + r"\\4_Substraction-TestImages\\"

noImage = 1
anotherParams = 1
imagesName = Rotate + str(noImage) + "." + str(anotherParams)

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("folderSubstractionImages: ", folderSubstractionImages)
print("folderCropImages: ", folderCropImages)
print("noImage: ", noImage)
print("anotherParams: ", anotherParams)
print("imagesName: ", imagesName)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# Process Images
for filename in os.listdir(folderCropImages):
    if filename[:5] == imagesName:
        print(filename)
        imagePath = folderCropImages + filename
        if filename[6:7] == "A":
            imgSubsA = cv2.imread(imagePath, 0)
            cv2.imshow('Original Image ' + str(filename), imgSubsA)
        elif filename[6:7] == "B":
            imgSubsB = cv2.imread(imagePath, 0)
            cv2.imshow('Original Image ' + str(filename), imgSubsB)
        elif filename[6:7] == "C":
            imgSubsC = cv2.imread(imagePath, 0)
            cv2.imshow('Original Image ' + str(filename), imgSubsC)
        elif filename[6:7] == "D":
            imgSubsD = cv2.imread(imagePath, 0)
            cv2.imshow('Original Image ' + str(filename), imgSubsD)

cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images A
A_A = imgSubsA - imgSubsA
A_B = imgSubsA - imgSubsB
A_C = imgSubsA - imgSubsC
A_D = imgSubsA - imgSubsD

cv2.imshow("A-A", A_A)
cv2.imshow("A-B", A_B)
cv2.imshow("A-C", A_C)
cv2.imshow("A-D", A_D)
cv2.imwrite(folderSubstractionImages + filename[:6] + "A-A-Subs.jpg", A_A)
cv2.imwrite(folderSubstractionImages + filename[:6] + "A-B-Subs.jpg", A_B)
cv2.imwrite(folderSubstractionImages + filename[:6] + "A-C-Subs.jpg", A_C)
cv2.imwrite(folderSubstractionImages + filename[:6] + "A-D-Subs.jpg", A_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images B
B_A = imgSubsB - imgSubsA
B_B = imgSubsB - imgSubsB
B_C = imgSubsB - imgSubsC
B_D = imgSubsB - imgSubsD

cv2.imshow("B-A", B_A)
cv2.imshow("B-B", B_B)
cv2.imshow("B-C", B_C)
cv2.imshow("B-D", B_D)
cv2.imwrite(folderSubstractionImages + filename[:6] + "B-A-Subs.jpg", B_A)
cv2.imwrite(folderSubstractionImages + filename[:6] + "B-B-Subs.jpg", B_B)
cv2.imwrite(folderSubstractionImages + filename[:6] + "B-C-Subs.jpg", B_C)
cv2.imwrite(folderSubstractionImages + filename[:6] + "B-D-Subs.jpg", B_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images C
C_A = imgSubsC - imgSubsA
C_B = imgSubsC - imgSubsB
C_C = imgSubsC - imgSubsC
C_D = imgSubsC - imgSubsD

cv2.imshow("C-A", C_A)
cv2.imshow("C-B", C_B)
cv2.imshow("C-C", C_C)
cv2.imshow("C-D", C_D)
cv2.imwrite(folderSubstractionImages + filename[:6] + "C-A-Subs.jpg", C_A)
cv2.imwrite(folderSubstractionImages + filename[:6] + "C-B-Subs.jpg", C_B)
cv2.imwrite(folderSubstractionImages + filename[:6] + "C-C-Subs.jpg", C_C)
cv2.imwrite(folderSubstractionImages + filename[:6] + "C-D-Subs.jpg", C_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images D
D_A = imgSubsD - imgSubsA
D_B = imgSubsD - imgSubsB
D_C = imgSubsD - imgSubsC
D_D = imgSubsD - imgSubsD

cv2.imshow("D-A", D_A)
cv2.imshow("D-B", D_B)
cv2.imshow("D-C", D_C)
cv2.imshow("D-D", D_D)
cv2.imwrite(folderSubstractionImages + filename[:6] + "D-A-Subs.jpg", D_A)
cv2.imwrite(folderSubstractionImages + filename[:6] + "D-B-Subs.jpg", D_B)
cv2.imwrite(folderSubstractionImages + filename[:6] + "D-C-Subs.jpg", D_C)
cv2.imwrite(folderSubstractionImages + filename[:6] + "D-D-Subs.jpg", D_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

Number Test:  1
Rotate True
folderSubstractionImages:  _Result Calibration_\\Test 1\\4_Substraction-TestImages\\
folderCropImages:  _Result Calibration_\\Test 1\\3_Crop-Images (Final)\\
noImage:  1
anotherParams:  1
imagesName:  R.1.1
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

R.1.1-A-Final.jpg
R.1.1-B-Final.jpg
R.1.1-C-Final.jpg
R.1.1-D-Final.jpg


*** Test Images - Substraction (grayscale-equalizeHist)***
===================================================

In [33]:
# Substraction grayscale-equalizeHist

import numpy as np
import cv2

numberTest = 1
Rotate = "R."    # if no Ratate, change to ""
folderCropImages = r"_Result Calibration_\\Test " + str(numberTest) + r"\\3_Crop-Images (Final)\\"
folderSubstractionImagesCustom = r"_Result Calibration_\\Test " + str(numberTest) + r"\\4_Substraction-TestImages-Custom\\"

noImage = 1
anotherParams = 1
imagesName = Rotate + str(noImage) + "." + str(anotherParams)

#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
print("Number Test: ", numberTest)
if Rotate == "R.":
    print("Rotate True")
else:
    print("Rotate False")
print("folderSubstractionImagesCustom: ", folderSubstractionImagesCustom)
print("folderCropImages: ", folderCropImages)
print("noImage: ", noImage)
print("anotherParams: ", anotherParams)
print("imagesName: ", imagesName)
print(">"*50)
print()
#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

# Process Images
for filename in os.listdir(folderCropImages):
    if filename[:5] == imagesName:
        print(filename)
        imagePath = folderCropImages + filename
        if filename[6:7] == "A":
            imgSubsA = cv2.imread(imagePath, 0)
            # Histogram Equalization
            imgSubsA = cv2.equalizeHist(imgSubsA)
            cv2.imshow('Original Image ' + str(filename), imgSubsA)
        elif filename[6:7] == "B":
            imgSubsB = cv2.imread(imagePath, 0)
            # Histogram Equalization
            imgSubsB = cv2.equalizeHist(imgSubsB)
            cv2.imshow('Original Image ' + str(filename), imgSubsB)
        elif filename[6:7] == "C":
            imgSubsC = cv2.imread(imagePath, 0)
            # Histogram Equalization
            imgSubsC = cv2.equalizeHist(imgSubsC)
            cv2.imshow('Original Image ' + str(filename), imgSubsC)
        elif filename[6:7] == "D":
            imgSubsD = cv2.imread(imagePath, 0)
            # Histogram Equalization
            imgSubsD = cv2.equalizeHist(imgSubsD)
            cv2.imshow('Original Image ' + str(filename), imgSubsD)

cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images A
A_A = imgSubsA - imgSubsA
A_B = imgSubsA - imgSubsB
A_C = imgSubsA - imgSubsC
A_D = imgSubsA - imgSubsD

cv2.imshow("A-A", A_A)
cv2.imshow("A-B", A_B)
cv2.imshow("A-C", A_C)
cv2.imshow("A-D", A_D)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "A-A-Subs-Cstm.jpg", A_A)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "A-B-Subs-Cstm.jpg", A_B)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "A-C-Subs-Cstm.jpg", A_C)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "A-D-Subs-Cstm.jpg", A_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images B
B_A = imgSubsB - imgSubsA
B_B = imgSubsB - imgSubsB
B_C = imgSubsB - imgSubsC
B_D = imgSubsB - imgSubsD

cv2.imshow("B-A", B_A)
cv2.imshow("B-B", B_B)
cv2.imshow("B-C", B_C)
cv2.imshow("B-D", B_D)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "B-A-Subs-Cstm.jpg", B_A)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "B-B-Subs-Cstm.jpg", B_B)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "B-C-Subs-Cstm.jpg", B_C)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "B-D-Subs-Cstm.jpg", B_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images C
C_A = imgSubsC - imgSubsA
C_B = imgSubsC - imgSubsB
C_C = imgSubsC - imgSubsC
C_D = imgSubsC - imgSubsD

cv2.imshow("C-A", C_A)
cv2.imshow("C-B", C_B)
cv2.imshow("C-C", C_C)
cv2.imshow("C-D", C_D)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "C-A-Subs-Cstm.jpg", C_A)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "C-B-Subs-Cstm.jpg", C_B)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "C-C-Subs-Cstm.jpg", C_C)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "C-D-Subs-Cstm.jpg", C_D)
cv2.waitKey(0)
cv2.destroyAllWindows()

# Substraction Images D
D_A = imgSubsD - imgSubsA
D_B = imgSubsD - imgSubsB
D_C = imgSubsD - imgSubsC
D_D = imgSubsD - imgSubsD

cv2.imshow("D-A", D_A)
cv2.imshow("D-B", D_B)
cv2.imshow("D-C", D_C)
cv2.imshow("D-D", D_D)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "D-A-Subs-Cstm.jpg", D_A)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "D-B-Subs-Cstm.jpg", D_B)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "D-C-Subs-Cstm.jpg", D_C)
cv2.imwrite(folderSubstractionImagesCustom + filename[:6] + "D-D-Subs-Cstm.jpg", D_D)
cv2.waitKey(0)
cv2.destroyAllWindows()


Number Test:  1
Rotate True
folderSubstractionImagesCustom:  _Result Calibration_\\Test 1\\4_Substraction-TestImages-Custom\\
folderCropImages:  _Result Calibration_\\Test 1\\3_Crop-Images (Final)\\
noImage:  1
anotherParams:  1
imagesName:  R.1.1
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

R.1.1-A-Final.jpg
R.1.1-B-Final.jpg
R.1.1-C-Final.jpg
R.1.1-D-Final.jpg
